glaring-client/src/components/Application.jsx

```diff
@@ -1,7 +1,23 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./Application.scss";
+import axios from "axios";
+
 import Sidebar from "./Sidebar/Index";
 
+
 export default function Application() {
+
+const [state, setState] = useState("smoko");
+  const [groups, setGroups] = useState([]);
+
+useEffect(() => {
+    axios.get(`/api/groups`)  
+      .then((response) => {
+
+        console.log("AAA", response)
+        setGroups(()=> response.data);
+      });
+  }, [state])
```

Comment:
having a self-referring dependency here is going to cause an infinite rendering loop.

the useEffect will get triggered by the state change and will change the state again and trigger itself and so on